metal/watchers.go

```diff
@@ -0,0 +1,143 @@
+package metal
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+)
+
+// createNodesWatcher create a cache.SharedIndexInformation for node handlers
+func createNodesWatcher(ctx context.Context, informer informers.SharedInformerFactory, cloudServices []cloudService) (cache.SharedIndexInformer, error) {
```

Comment:
If we were to implement `LoadBalancer`, we would get these node/service watchers/informers for free. 
I don't know if kube-vip or metallb register themselves as cloud-provider LoadBalancers (what does that really mean? I don't know.), I presume they do not.  If they do I would be curious to know what users would provider for Service loadBalancerClass https://kubernetes.io/docs/concepts/services-networking/service/#load-balancer-class.

Why do we not implement LoadBalancer?

Reply:
We should. It is on my list of todos, to which I haven't gotten yet. Your work on `InstancesV2()` also inspired me to think about getting moving on that. If it simplifies our work, even better.

Reply:
But, like moving `watchers.go` into an internal package, subject for another PR.